Source/WebKit/NetworkProcess/curl/NetworkDataTaskCurl.cpp

```diff
@@ -61,17 +61,18 @@ NetworkDataTaskCurl::NetworkDataTaskCurl(NetworkSession& session, NetworkDataTas
     , m_sourceOrigin(parameters.sourceOrigin)
 {
     auto request = parameters.request;
-    if (request.url().protocolIsInHTTPFamily()) {
-        if (m_storedCredentialsPolicy == StoredCredentialsPolicy::Use) {
-            auto url = request.url();
-            m_user = url.user();
-            m_password = url.password();
-            request.removeCredentials();
-
+    auto url = request.url();
+    if (m_storedCredentialsPolicy == StoredCredentialsPolicy::Use && url.protocolIsInHTTPFamily()) {
+        m_user = url.user();
+        m_password = url.password();
+        request.removeCredentials();
+        url = request.url();
```

Comment:
Why do you do `url = request.url();` twice here?

Reply:
This is to get the url after the credential has been removed with request.removeCredentials().

https://github.com/WebKit/WebKit/blob/74a570d11375304d6dd2fe7e2e2696e5f93bcff2/Source/WebCore/platform/network/ResourceRequestBase.cpp#L190-L199
=> `m_requestData.m_url.removeCredentials();` removes credentials in the URL.

https://github.com/WebKit/WebKit/blob/74a570d11375304d6dd2fe7e2e2696e5f93bcff2/Source/WebCore/platform/network/ResourceRequestBase.cpp#L121-L126
=> Then get the URL with the credential removed.

Reply:
This process is exactly the same as the cocoa port. Also, the logic itself remains the same as before the change.

Reply:
Got it